### PR TITLE
feat(js-sdk): add Node.js SDK

### PR DESCRIPTION
**Pull request by fengmk2**
**What this PR does**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/mosn/layotto/issues/258

**Special notes for your reviewer**:

- [x] State API
- [x] File API
- [x] Sequencer API
- [x] Distributed Lock API
- [x] Pub/Sub API
- [x] RPC API
- [x] Binding API

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by fengmk2** [comment]
@seeflood 所有已知的 API 都实现了，可以 review 了。

**Comment by seeflood** [comment]
Thanks for your contribution!
I will modify the check license configuration and add license headers once the code review finished (to avoid be overwrited by your push) So ignore the CI warning for now :)

File: configs/config_integration_redis_etcd.json [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+                          }
+                        },
+                        "redis": {
+                          "metadata": {
+                            "redisHost": "localhost:6380",
+                            "redisPassword": ""
+                          }
+                        }
+                      },
+                      "pub_subs": {
```
Comment (fengmk2) [inline configs/config_integration_redis_etcd.json:126]: 这里配置没有生效

File: spec/proto/runtime/v1/runtime.proto [hunk 2]
```diff
@@ -163,7 +163,8 @@ message SequencerOptions {
 
 message GetNextIdResponse{
   // The next unique id
-  int64 next_id = 1;
+  // Fixed int64 overflow problems on JavaScript https://github.com/improbable-eng/ts-protoc-gen#gotchas
+  int64 next_id = 1 [jstype = JS_STRING];
```
[the remark this answers is not in the document]
Comment (seeflood) [reply spec/proto/runtime/v1/runtime.proto:167]: ok

**Comment by fengmk2** [comment]
move into https://github.com/layotto/js-sdk

**Comment by seeflood** [comment]
Hi @fengmk2 ,it seems that we still need part of your PR, like:
- jstype in runtime.proto
image

- configs/config_integration_redis_etcd.json 
- Makefile
image

Could u please submit a PR only including these ? Or should I just reopen and modify this PR

**Comment by fengmk2** [comment]
@seeflood new PR landed https://github.com/mosn/layotto/pull/316

**Pull request closed without merging**